A32NX/html_ui/Pages/A32NX_Core/A32NX_APU.js

```diff
@@ -4,72 +4,64 @@ class A32NX_APU {
-            const apuStartButtonPressed = SimVar.GetSimVarValue("L:A32NX_APU_START_ACTIVATED", "Bool");
-            if (apuFuelsystemValveOpen === 100 && apuStartButtonPressed) {
-                // This fires the APU_STARTER key event, which will cause `A:APU SWITCH` to be set to 1
-                SimVar.SetSimVarValue("K:APU_STARTER", "Number", 1);
-            }
+        // Until everything that depends on the APU is moved into WASM,
+        // we still need to synchronise some of the WASM state with the sim's state.
+        if (available && !apuSwitchIsOn) {
+            // This event will open the fuel valve leading to the APU.
+            SimVar.SetSimVarValue("K:FUELSYSTEM_VALVE_TOGGLE", "Number", 8);
+            // This event will set `A:APU SWITCH` to 1, meaning the sim will start the APU.
+            // In systems.cfg, the `apu_pct_rpm_per_second` setting is set to 1, meaning the APU starts in one second.
+            SimVar.SetSimVarValue("K:APU_STARTER", "Number", 1);
+        } else if (!available && apuSwitchIsOn) {
+            // This event will set `A:APU SWITCH` to 0, meaning the sim will stop the APU.
+            // In systems.cfg, the `apu_pct_rpm_per_second` setting is set to 1, meaning the APU stops in one second.
+            SimVar.SetSimVarValue("K:APU_OFF_SWITCH", "Number", 1);
+            // This event will close the fuel valve leading to the APU.
+            SimVar.SetSimVarValue("K:FUELSYSTEM_VALVE_TOGGLE", "Number", 8);
         }
 
-        // Takes 20 seconds to open
-        const apuFlapOpenPercentSpeed = 20;
+        const apuGenActive = SimVar.GetSimVarValue("APU GENERATOR ACTIVE", "Bool");
+        const externalPowerOff = SimVar.GetSimVarValue("EXTERNAL POWER ON", "Bool") === 0;
 
-        if (currentAPUMasterState === 1 && apuFlapOpenPercent < 100) {
-            const newFlap = Math.min(apuFlapOpenPercent + ((100 / apuFlapOpenPercentSpeed) * (_deltaTime / 1000)), 100);
-            SimVar.SetSimVarValue("L:APU_FLAP_OPEN", "Percent", newFlap);
-        } else if (currentAPUMasterState === 0 && apuFlapOpenPercent > 0 && APUPctRPM <= 7) {
-            const newFlap = Math.max(apuFlapOpenPercent - ((100 / apuFlapOpenPercentSpeed) * (_deltaTime / 1000)), 0);
-            SimVar.SetSimVarValue("L:APU_FLAP_OPEN", "Percent", newFlap);
-        }
+        // This logic is consistently faulty in the JavaScript code: of course it should also take into
+        // account if engine generators are supplying electricity. We'll fix this when we create the electrical system.
+        SimVar.SetSimVarValue("L:APU_GEN_ONLINE", "Bool", available && apuGenActive ? 1 : 0);
+        SimVar.SetSimVarValue(
+            "L:APU_LOAD_PERCENT",
+            "percent",
+            available && apuGenActive && externalPowerOff
+                ? Math.max(SimVar.GetSimVarValue("L:A32NX_APU_GEN_AMPERAGE", "Amperes")
+                        / SimVar.GetSimVarValue("ELECTRICAL TOTAL LOAD AMPS", "Amperes"), 0)
```

Comment:
This looks very weird on first glance, but if you cross your eyes and consider how limited the sim's electrical simulation is it seems okay. At face value with conventional terminology the load percentage would be A32NX_APU_GEN_AMPERAGE / \<APU gen rated current\>.